### ISSUE TITLE
Broken images being fetched

### ISSUE THREAD

**Issue by UndeadKernel** [maubot/xkcd#2]
Hey, it seems that the bot is fetching broken PNG images.
No matter if in the config I have the comics set to display inline or not, the comics are not correctly displayed. It seems that garbage bytes are being added to the fetched PNG file.

**Comment by tulir** [comment]
The problem might be not having `python-magic` installed, and thus the bot sending the wrong mime type when uploading the file.

**Comment by UndeadKernel** [comment]
Thanks for the suggestions @tulir. 
After installing `python-magic` (in the maubot python environment), I still have the same issue.

The image is uploaded correctly to the matrix media folder and I can see it correctly with a image viewing program.

Any clue how I could debug this issue?

**Comment by mozzieongit** [comment]
Hi, I had the same issue. 
It seemed like the bot was uploading the PNG image as JSON data. I did a curl on the image uploaded by the bot:
```
HTTP/1.1 200 OK
[...]
Content-Type: application/json; charset=UTF-8
Content-Length: 51409
[...]
```

curling a different picture (uploaded by me) results in the correct content-type: image/png

This only happened on the image sent when using `!xkcd` or `!xkcd 2497` (as the current number) to get the latest image. When using `!xkcd 1500` then the resulting image was displayed correctly.
I found that the row in media_cache of the bot database had an empty cell for mime_type.
I then deleted the media_cache row containing the file_name logic_gates.png (the broken image).
`delete FROM "media_cache" where file_name = 'logic_gates.png'`
After deleting the row I re-triggered the bot with `!xkcd` and the image loaded fine.

NOTE: I installed python-magic mid-way through, so I don't remember if the second image was loaded correctly because of that. The broken image was still broken after that until I deleted the cache entry.

TL;DR: installed python-magic, deleted broken images (empty mime_type) from bot database via the web management tool.
